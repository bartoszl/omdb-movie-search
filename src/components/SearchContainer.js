import styled from 'styled-components';
import { BACKGROUND_GREY } from '../constants/colors';

const SearchContainer = styled.div`
  width: 100vw;
  height: ${({ collapsed }) => (collapsed ? '15vh' : '100vh')};
  background-color: ${BACKGROUND_GREY};
  display: flex;
  align-items: center;
  justify-content: center;
  transition: height 0.75s ease-out;
  position: fixed;
  top: 0;
  left: 0;
`;

export default SearchContainer;
