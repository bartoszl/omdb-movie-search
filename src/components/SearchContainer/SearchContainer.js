import styled from 'styled-components';
import { BACKGROUND_GREY } from '../../constants/colors';
import bg from '../../assets/search-bg.jpg';

const SearchContainer = styled.div`
  width: 100vw;
  height: ${({ collapsed }) => (collapsed ? '15vh' : '100vh')};
  background-image: ${({ collapsed }) => (!collapsed ? `url(${bg})` : 'none')};
  background-color: ${BACKGROUND_GREY};
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  transition: all 0.75s ease-out;
  position: fixed;
  top: 0;
  left: 0;
`;

export default SearchContainer;
