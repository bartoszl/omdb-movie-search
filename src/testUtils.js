import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
