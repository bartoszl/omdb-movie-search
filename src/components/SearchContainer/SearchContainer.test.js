import React from 'react';
import { render, cleanup } from '../../testUtils';
import SearchContainer from './SearchContainer';

describe('InfoLabel', () => {
  afterEach(() => {
    cleanup();
  });

  test('matches snapshot', () => {
    const { asFragment } = render((
      <SearchContainer />
    ));

    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot collapsed', () => {
    const { asFragment } = render((
      <SearchContainer collapsed />
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
