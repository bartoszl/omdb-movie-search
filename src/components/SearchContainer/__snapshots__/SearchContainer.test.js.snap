// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLabel matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kMIWSX"
  />
</DocumentFragment>
`;

exports[`InfoLabel matches snapshot collapsed 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx dZKstf"
  />
</DocumentFragment>
`;
